core/controllers/resources_test.py

```diff
@@ -133,3 +132,45 @@ def test_bad_filenames_are_detected(self):
         self.assertIn('Filenames should not include', response_dict['error'])
 
         self.logout()
+
+    def test_missing_extensions_are_detected(self):
```

Comment:
Can you add a test for test_images_with_unexpected_extensions_are_still_accepted() to test the "false negative" case you mentioned, where a file was an image but of a different type than detected? I think that would be useful for future reference especially if devs try to change the validation again in the future.